cyhy/db/ticket_manager.py

```diff
@@ -108,13 +109,54 @@ def __generate_ticket_details(self, vuln, ticket, check_for_changes=True):
             cve_doc = self.__db.CVEDoc.find_one({"_id": vuln["cve"]})
             if cve_doc:
                 new_details["cvss_base_score"] = cve_doc["cvss_score"]
+                new_details["cvss_version"] = cve_doc["cvss_version"]
                 new_details["score_source"] = "nvd"
                 new_details["severity"] = cve_doc["severity"]
             # if the CVE is listed in the KEV collection, we'll mark it as such
             kev_doc = self.__db.KEVDoc.find_one({"_id": vuln["cve"]})
             if kev_doc:
                 new_details["kev"] = True
 
+        # As of May 2022, some Nessus plugins report a severity that is
+        # inconsistent with their (non-NVD, non-CVE-based) CVSS v3 score.
+        # To reduce confusion, we ensure that the severity is correct here.
+        # For examples, see the following plugins:
+        # 34460, 104572, 107056, 140770, 156560, 156941, 156441
+        if new_details["score_source"] != "nvd":
+            cvss = new_details["cvss_base_score"]
+            # Source: https://nvd.nist.gov/vuln-metrics/cvss
+            #
+            # Notes:
+            # - The CVSS score to severity mapping is not continuous (e.g. a
+            #   score of 8.95 is undefined according to their table).
+            #   However, the CVSS equation documentation
+            #   (https://www.first.org/cvss/specification-document#CVSS-v3-1-Equations)
+            #   specifies that all CVSS scores are rounded up to the nearest
+            #   tenth of a point, so our severity mapping below is valid.
+            # - CVSSv3 specifies that a score of 0.0 has a severity of "None",
+            #   but we have chosen to map 0.0 to severity 1 ("Low") because
+            #   CyHy code has historically assumed severities between 1 and 4
+            #   (inclusive).  Since we have not seen CVSSv3 scores lower than
+            #   3.1, this will hopefully never be an issue.
+            if new_details["cvss_version"] == "2":
```

Comment:
I'm not quite sure I totally understand your question, but the basic idea here is that a CyHy ticket only gets a "real" CVSS version string (e.g. `[2.0, 3.0, 3.1]`) if there is an actual CVE associated with the finding and that CVE exists in our CVEs collection.  Otherwise, we default to whatever CVSS version is reported by the vuln scanner, which is either `2` or `3`.  This allows us to easily tell whether a CVSS version came from the NVD or a vuln scanner vendor.  Yes, that info is also available in the `source` field of a ticket, but this gives us another way to easily make that distinction.  This mechanism was discussed with the CyHy team and agreed upon.

If this doesn't address your concern, let me know and I can try again.

Reply:
Sorry it was me missing the `new_details["score_source"] != "nvd"` check and worrying that the `cvss_version` string pulled from a CVEDoc is not the same format as the `if`/`elif` checks here. We also pull the `severity` from the CVEDoc if one is found so it's a non-issue anyway. Apologies for the confusion from me.

Reply:
No problem!  Can you resolve this convo?